File: source/SIL.AppBuilder.Portal.Frontend/src/data/models/product-build.ts
import { AttributesObject, ResourceObject } from 'jsonapi-typescript';

export type PRODUCT_BUILD_TYPE = 'productBuild';

export interface ProductBuildAttributes extends AttributesObject {
  version: string;
  success?: boolean;
  buildId: number;
  dateCreated: string;
  dateUpdated: string;
}

export type ProductBuildResource = ResourceObject<PRODUCT_BUILD_TYPE, ProductBuildAttributes>;

File: source/OptimaJet.DWKit.StarterApplication/Models/ProductBuild.cs
using System;
using System.Collections.Generic;
using JsonApiDotNetCore.Models;

namespace OptimaJet.DWKit.StarterApplication.Models
{
    public class ProductBuild : Identifiable, ITrackDate
    {
        [HasOne("product")]
        public virtual Product Product { get; set; }
        public Guid ProductId { get; set; }

        [Attr("build-id")]
        public int BuildId { get; set; }

        [Attr("version")]
        public string Version { get; set; }

        [Attr("success")]
        public bool? Success { get; set; }

        [Attr("date-created")]
        public DateTime? DateCreated { get; set; }

        [Attr("date-updated")]
        public DateTime? DateUpdated { get; set; }

        [HasMany("product-artifacts", Link.None)]
        public virtual List<ProductArtifact> ProductArtifacts { get; set; }

        [HasMany("product-publications", Link.None)]
        public virtual List<ProductPublication> ProductPublications { get; set; }
    }
}
